dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -58,7 +60,7 @@ class ReceivedDataSample;
 typedef RcHandle<RtpsUdpInst> RtpsUdpInst_rch;
 typedef RcHandle<RtpsUdpTransport> RtpsUdpTransport_rch;
 
-class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink {
+  class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink, public virtual NetworkConfigListener {
```

Comment:
I don't think this is the right place for this inheritance. The transport's datalink, as I understand it, is a connection between two transport instances. I think these listener callbacks would impact the transport at a higher level than the datalink, maybe even on the (RtpsUdp)TransportImpl itself. This feels like it needs some design discussion.

Reply:
In principle, I agree with you.  It should be a Transport level thing.  If we ever updated say the tcp transport to respond to network changes, then the Transport would be `NetworkConfigListener`.  With that being said however, each Transport is free to implement this however they choose.

For RTPS, the datalink has the information necessary to process network changes.  If the RtpsUdpTransport were the NetworkConfigListener, then it would just be dispatching down to the RtpsUdpDataLink.

Also, if you trace the call sequence then a network change goes from RtpsUdpDatalink -> Datalink -> TransportSendListeners and TransportReceiverListeners -> DataReaderImpl and DataWriterImpl -> Discovery.  I did not see anything in the framework where something higher than the datalink was keeping tracking of all the TransportClients.  If the TransportImpl maintained a list of TransportClients, then it would be a different story.  We could go from TransportImpl -> TransportClient -> Discovery.

So, my current plan is to leave it as is hoping that we can retire the transport framework before we need to make the aforementioned improvements.